### DISCUSSION TITLE
Custom Missions

### DISCUSSION THREAD

**Discussion by Iazu** [LostArtefacts/TRX#813] [Q&A] [answered]
I've been starting to delve into the custom missions available on TRCustoms and TRLE lately and so far the ones I've tried seem to be working just fine with Tomb1Main except for one tiny thing. Which is only an annoyance not at all game impacting. 

Here's what I've been doing..
Made a backup of DATA, FMV, music, and shaders folders. Then take the same folders out of the custom level's zip and copy them over the existing ones in the main Tomb Raider folder, overwriting everything. This seems to work even if it is a bit clunky and (imo) it's better than running the provided tombati.exe with custom levels. Then, I can just restore the backups when I want to go back to default levels.

The only issue I've noticed so far aside from having to do a bunch of file manipulation every time I load a new level is that the saves are named as if I'm playing the stock levels. So the first level's save is always named "Caves", etc etc. I'm guessing Tomb1Main is pulling all this info from the json files in the cfg folder.

So I guess I have two questions. Is there an easier way to go about loading in custom levels into Tomb1Main and is there a way for it to pull the actual name of the level for the saves aside from editing the json file?

**Comment by lahm86** [comment, marked as answer]
I think the only other way you could do this is by contacting the level authors to request updates to their TRLEs to target Tomb1Main - that would have the added benefit of authors testing their levels, but I can't comment on the sort of response you would get.

The level names are taken from the gameflow json: there is no other mechanism to do this in Tomb1Main. You could customize the gameflow for each TRLE prior to starting the game. In TRLEs that use tombati, I believe the way authors achieved custom level names was to manually edit the executable in a hex editor e.g.

![image](https://user-images.githubusercontent.com/33758420/233774283-266ffe36-45db-4b86-b8f7-3fd790acc47b.png)

So you could use a similar approach to extract the names prior to playing the game. Note that the names appear in reverse.

**Reply by Iazu** [reply]
Thanks for the reply! That's good to know. Didn't even think about them using something like hex editing to fool the game into doing that, clever! (also, the example you gave is coincidentally what I was playing when I asked the question. Just finished "The Lost Ship")

In the end it's a minor annoyance and it's easy enough to edit the gameflow json if I really want the saves to match. Might just do a separate install and use "Level 1, Level 2" etc just for a generic solution.

I *do* wish there was an easier way to load custom levels, like a mod/level loader or something like some other games have. But hey, this at least works and isn't that much of a hassle. Though It's actually kind of cool that TRLE packages each map set like it's own little standalone game.

**Reply by walkawayy** [reply]
Btw to simplify your setup, you can have multiple folders for Tomb1Main and they won't conflict with each other. That way you can avoid overwriting files every time you play a new custom level set or want to play the OG game.

For example:
`Downloads/Tomb1Main`
`Downloads/Nordic Adventure`
`Downloads/Sabatu TR1`
etc.
